cmd/konnect_sync.go

```diff
@@ -19,7 +21,9 @@ to get Konnect's state in sync with the input state.` + konnectAlphaState,
 			return fmt.Errorf("writing to stdout is not supported in Konnect mode")
 		}
 		_ = sendAnalytics("konnect-sync", "")
-		return syncKonnect(cmd.Context(), konnectDiffCmdKongStateFile, false,
+		ctx, cancel := context.WithTimeout(cmd.Context(), time.Second*time.Duration(rootConfig.Timeout))
+		defer cancel()
+		return syncKonnect(ctx, konnectDiffCmdKongStateFile, false,
```

Comment:
This seems incorrect. This would imply that the whole sync process should finish in rootconfig.Timeout.

rootConfig.Timetout is a request-level timeout. The cmd.context() seems right, which will be canceled on a signal.